### DISCUSSION TITLE
How to deploy Appwrite with multiple hard drives or servers?

### DISCUSSION THREAD

**Discussion by DankMemeGuy** [appwrite/appwrite#5000] [Q&A] [answered]
### 💭 Description

I know it's not something many people will necessarily encounter. However, with users uploading files, and database growing, etc. Is there any possibility of updating the documentation on how to scale AppWrite to use multiple hard drives or servers? Having an NVME dedicated to storage, or an NVME dedicated to database etc, and just diversifying the load, or even having multiple instances of AppWrite that can talk to each other and spread the compute/database lookups etc.

I think if people think more smaller scale too, if users run AppWrite on a small NVME like 100GB, then it grows, they either need to buy a larger NVME, or if they have a secondary slot they could add a new stick and they could get even more space without disposing of the old working NVME. Having AppWrite documentation or feature to allow multiple hard drives would be quite useful for a bootstrapped start-up.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment, marked as answer]
### How to scale Appwrite to use multiple hard drives

Appwrite uses Docker and, as such, uses volumes to persist the data. There should be resources online if you want to change how the volumes work, including specifying where the files should go. See [docker-compose reference](https://docs.docker.com/compose/compose-file/compose-file-v3/#volumes) for more info.

### Scaling the Database

At the moment, Appwrite connects to a MariaDB Database. Again, there are resources online for how to scale MariaDB. We actually recommend using a managed provider to make things easy.

### Scaling Servers

To scale the rest of the containers that actually do the work of handling API requests, we recommend using Docker Swarm. Maybe we'll document this more in the future.

**Reply by DankMemeGuy** [reply]
tahnk you!
